semver.js

```diff
@@ -793,6 +793,56 @@ function toComparators(range, loose) {
   });
 }
 
+exports.comparatorsIntersect = comparatorsIntersect;
+function comparatorsIntersect(compA, compB, loose, platform) {
+  compA = new Comparator(compA, loose);
+  compB = new Comparator(compB, loose);
```

Comment:
Reassigning function arguments deopts in v8; could you use new variables for these?